The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/19751096058/job/56594181986) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/56594181986) [(plain)](https://github.com/rust-lang/rust/commit/845da883816d60b6a3c8b161ecf5d3c306867afa/checks/56594181986/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
291 |     T: 'a + Ord,
    |             ^^^ required by this bound in `PeekMut`
help: consider restricting type parameter `T` with trait `Ord`
    |
540 | impl<T: core::cmp::Ord, A: Allocator> BinaryHeap<T, A> {
    |       ++++++++++++++++

[RUSTC-TIMING] core test:false 34.212
   Compiling memchr v2.7.6
error[E0277]: the trait bound `T: Ord` is not satisfied
   --> library/alloc/src/collections/binary_heap/mod.rs:616:30
    |
616 |         if self.is_empty() { None } else { Some(PeekMut { heap: self, original_len: None }) }
    |                              ^^^^ the trait `Ord` is not implemented for `T`
    |
note: required by a bound in `binary_heap::PeekMut`
   --> library/alloc/src/collections/binary_heap/mod.rs:291:13
    |
289 | pub struct PeekMut<
    |            ------- required by a bound in this struct
290 |     'a,
291 |     T: 'a + Ord,
    |             ^^^ required by this bound in `PeekMut`
help: consider restricting type parameter `T` with trait `Ord`
    |
540 | impl<T: core::cmp::Ord, A: Allocator> BinaryHeap<T, A> {
    |       ++++++++++++++++

error[E0277]: the trait bound `T: Ord` is not satisfied
   --> library/alloc/src/collections/binary_heap/mod.rs:616:65
    |
616 |         if self.is_empty() { None } else { Some(PeekMut { heap: self, original_len: None }) }
    |                                                                 ^^^^ the trait `Ord` is not implemented for `T`
    |
note: required by a bound in `binary_heap::PeekMut`
   --> library/alloc/src/collections/binary_heap/mod.rs:291:13
    |
289 | pub struct PeekMut<
    |            ------- required by a bound in this struct
290 |     'a,
291 |     T: 'a + Ord,
    |             ^^^ required by this bound in `PeekMut`
help: consider restricting type parameter `T` with trait `Ord`
    |
540 | impl<T: core::cmp::Ord, A: Allocator> BinaryHeap<T, A> {
    |       ++++++++++++++++

error[E0277]: the trait bound `T: Ord` is not satisfied
   --> library/alloc/src/collections/binary_heap/mod.rs:616:44
    |
616 |         if self.is_empty() { None } else { Some(PeekMut { heap: self, original_len: None }) }
    |                                            ^^^^ the trait `Ord` is not implemented for `T`
    |
note: required by a bound in `binary_heap::PeekMut`
   --> library/alloc/src/collections/binary_heap/mod.rs:291:13
    |
289 | pub struct PeekMut<
    |            ------- required by a bound in this struct
290 |     'a,
291 |     T: 'a + Ord,
    |             ^^^ required by this bound in `PeekMut`
help: consider restricting type parameter `T` with trait `Ord`
    |
540 | impl<T: core::cmp::Ord, A: Allocator> BinaryHeap<T, A> {
    |       ++++++++++++++++

error[E0277]: the trait bound `T: Ord` is not satisfied
   --> library/alloc/src/collections/binary_heap/mod.rs:616:44
    |
616 |         if self.is_empty() { None } else { Some(PeekMut { heap: self, original_len: None }) }
    |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Ord` is not implemented for `T`
    |
note: required by a bound in `binary_heap::PeekMut`
   --> library/alloc/src/collections/binary_heap/mod.rs:291:13
    |
289 | pub struct PeekMut<
    |            ------- required by a bound in this struct
290 |     'a,
291 |     T: 'a + Ord,
    |             ^^^ required by this bound in `PeekMut`
help: consider restricting type parameter `T` with trait `Ord`
    |
540 | impl<T: core::cmp::Ord, A: Allocator> BinaryHeap<T, A> {
    |       ++++++++++++++++

[RUSTC-TIMING] libc test:false 1.908
   Compiling unwind v0.0.0 (/checkout/library/unwind)
[RUSTC-TIMING] unwind test:false 0.071
```

</details>

--- a/library/alloc/src/collections/binary_heap/mod.rs
+++ b/library/alloc/src/collections/binary_heap/mod.rs
@@ -581,7 +581,9 @@ impl<T, A: Allocator> BinaryHeap<T, A> {
     pub fn with_capacity_in(capacity: usize, alloc: A) -> BinaryHeap<T, A> {
         BinaryHeap { data: Vec::with_capacity_in(capacity, alloc) }
     }
+}
 
+impl<T: Ord, A: Allocator> BinaryHeap<T, A> {
     /// Returns a mutable reference to the greatest item in the binary heap, or
     /// `None` if it is empty.
     ///
@@ -615,9 +617,7 @@ impl<T, A: Allocator> BinaryHeap<T, A> {
     pub fn peek_mut(&mut self) -> Option<PeekMut<'_, T, A>> {
         if self.is_empty() { None } else { Some(PeekMut { heap: self, original_len: None }) }
     }
-}
 
-impl<T: Ord, A: Allocator> BinaryHeap<T, A> {
     /// Removes the greatest item from the binary heap and returns it, or `None` if it
     /// is empty.
     ///

